### ISSUE TITLE
Dropout layer has a poor perfomance

### ISSUE THREAD

**Issue by emartinezs44** [intel-analytics/BigDL#4867]
The **com.intel.analytics.bigdl.dllib.keras.layers.Dropout** layer is slow at the forward execution. Doing some experiments it takes about 0.2 seconds with an input of Tensor[Float](1, 12, 512, 112) shape (storage 3145728). The backward phase is like ten times faster. Watching the implementation I can see that the Bernoulli function is being executed for every single element of the Storage. This layer represents an important issue in the forward phase and should be as fast as the backward step.

**Comment by qiuxin2012** [comment]
I just run the performance of forwarding a Tensor(1, 12, 512, 512), it will cost 0.15s on my desktop, and the bernoulli for each elements cost 0.11 seconds.
Dropout masks parts of input using a bernoulli distribution, we should run a bernoulli for each elements to determine if we drop the element during the forward phase. The backward phase is much easier, just one operation gradInput * noise. So the backward is very fast. It's imposible to run forward phase as faster as backward phase, but we can try to improve the performance of bernoulli, we will info you if we make progess.

**Comment by emartinezs44** [comment]
Even 0.15 seconds is a lot. Note that transformer layers stacks 3 dropout nodes per block. So you have only in dropout almost 0.45 secs. If you consider 12 layers it causes several seconds of additional latency. The problem as I see is that there must be a better way to apply bernoulli. I will check other implementations in other frameworks to see how could it be improved.

**Comment by emartinezs44** [comment]
The forward phase in pytorch is about 0.01 secs in my laptop. The Dropout from com.intel.analytics.bigdl.dllib.nn is 0.15 secs in a Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz

**Comment by emartinezs44** [comment]
Doing some tests with bernoulli tensor implementation from Torch using bytedeco presets with openblas takes 0.033 secs. The same with bernoulli from bigdl Tensor takes 0.12 secs for a input tensor of shape  (1, 12, 512, 512).

sbt config:

val javacpp = "1.5.7"
lazy val pyTorchPresets = "org.bytedeco" % "pytorch-platform" % s"1.10.2-$javacpp"
lazy val torchCPU = "org.bytedeco" % "pytorch" % s"1.10.2-$javacpp"
lazy val openblas = "org.bytedeco" % "openblas"  % s"0.3.19-$javacpp"
lazy val openblas_platform =  "org.bytedeco" % "openblas-platform" % s"0.3.19-$javacpp"

Code:

import org.bytedeco.pytorch.IValue
val tensor = (new IValue(ones(1, 12, 512, 512))).toTensor
tensor.bernoulli(0.5)

**Comment by qiuxin2012** [comment]
I try to improve the performance of Dropout, see #4891 for the changes.

Time cost of tensor.bernoulli(0.5) is between 0.03s - 0.04s now. The total forward time reduced from 0.15s to 0.06s - 0.07s on my desktop.

**Comment by emartinezs44** [comment]
Good job!. My model takes 4 secods less now in forward phase. I see that the Futures are not being used, so maybe it would be better to remove the line that creates them.
